pkg/search/parser_test.go

```diff
@@ -51,3 +51,70 @@ func TestSplitQuery(t *testing.T) {
 		})
 	}
 }
+
+func TestParsePairs(t *testing.T) {
+	cases := []struct {
+		query      string
+		allowEmpty bool
+		pairs      []string
+		valid      bool
+	}{
+		{
+			query:      "Deployment Label:label",
+			allowEmpty: false,
+			pairs:      []string{"Deployment Label", "label"},
+			valid:      true,
+		},
+		{
+			query:      "Deployment Label:label,label",
+			allowEmpty: false,
+			pairs:      []string{"Deployment Label", "label,label"},
+			valid:      true,
+		},
+		{
+			query:      "  Deployment Label :  label+label  ",
+			allowEmpty: false,
+			pairs:      []string{"Deployment Label", "label+label"},
+			valid:      true,
+		},
+		{
+			query:      "Deployment Label:label+label",
+			allowEmpty: false,
+			pairs:      []string{"Deployment Label", "label+label"},
+			valid:      true,
+		},
+		{
+			query:      "Deployment Label",
+			allowEmpty: false,
+			pairs:      []string{"", ""},
+			valid:      false,
+		},
+		{
+			query:      "Deployment Label:",
+			allowEmpty: false,
+			pairs:      []string{"", ""},
+			valid:      false,
+		},
+		{
+			query:      "Deployment Label:",
+			allowEmpty: true,
+			pairs:      []string{"Deployment Label", WildcardString},
+			valid:      true,
+		},
```

Comment:
I'd also be interested in the outcome of `query` "Deployments Label" with `allowEmpty` set to true.

Reply:
Thanks for updating the test!